Fix paddings and margins in ChangeNeuronVisibilityForm

coskucinkilic opened the pull request:
# Motivation

There has been differences between figma design and ChangeNeuronVisibilityForm paddings, this pr aims to remove the difference and apply figma design as close as possible.

# Changes

1. Update ChangeBulkVisibilityForm and NeuronVisibililityRow paddings and margins

# Tests
before:
<img width="616" alt="Screenshot 2024-10-23 at 20 50 13" src="https://github.com/user-attachments/assets/e4b3a78b-ff34-49d8-b260-90d7f3b5122b">

after:
<img width="606" alt="Screenshot 2024-10-23 at 20 49 47" src="https://github.com/user-attachments/assets/3071ae77-152c-41f8-b9e2-45070723cea2">


# Todos

- [ ] Add entry to changelog (if necessary).

dskloetd commented on frontend/src/lib/modals/neurons/NeuronVisibilityRow.svelte:
```diff
@@ -78,18 +78,18 @@
       --elements-badges: var(--text-description-tint);
     }
+
+    :global(.checkbox) {
```
I missed this before. Why do we need to use `:global`?

coskucinkilic replied:
Without it i wasn't able to stretch the checkbox to full with and ended up with this.

<img width="513" alt="Screenshot 2024-10-23 at 16 52 04" src="https://github.com/user-attachments/assets/ada74df4-0b27-41d3-b000-d613d89c6a95">

dskloetd replied:
I think there should be a better way. Can you ask @peterpeterparker how he would do it?

coskucinkilic replied:
I have realised the component display flex was causing this issue, which is no longer needed so i have deleted both flex and global.

dskloetd commented on frontend/src/lib/modals/neurons/NeuronVisibilityRow.svelte:
```diff
@@ -103,11 +103,16 @@
     }
 
+    .neuron-id {
+      margin: 0;
```
Why use `<p>` if we don't want margins? Can we just use `<div>` instead?

coskucinkilic replied:
Because they are semantically different

dskloetd replied:
But then why use `<p>` in the first place.
`<p>` is semantically a paragraph, isn't it?
And a neuron ID isn't really a paragraph.
Am I missing something?

coskucinkilic replied:
Changed

coskucinkilic commented:
The separator was cut from left and right due to padding at the top level, i have changed the location of padding to be inside lists now, also realised a top padding difference for uncontrolled neruons so i added that in.

mstrasinskis reviewed (approved):
Nice, the filter checkboxes are fixed as well. Thank you!

<img width="475" alt="image" src="https://github.com/user-attachments/assets/8c6038ed-30e3-459a-b499-f07088e5440d">
